src/azul/service/repository_controller.py

```diff
@@ -51,13 +53,29 @@ def download_file(self,
         plugin = self.repository_plugin(catalog)
         download_cls = plugin.file_download_class()
 
+        is_completed_range_request = False
+
         if request_index == 0:
             file = self.service.get_data_file(catalog=catalog,
                                               file_uuid=file_uuid,
                                               file_version=file_version)
             if file is None:
                 raise NotFoundError(f'Unable to find file {file_uuid!r}, '
                                     f'version {file_version!r} in catalog {catalog!r}')
+            range_specifier = headers.get('range')
+            if range_specifier is not None:
+                units, _, range_set = range_specifier.partition('=')
+                if units == 'bytes':
+                    if range_set == '':
+                        raise BadRequestError(f'Invalid range specifier {range_specifier!r}')
+                    elif range_set.count(',') == 0:
+                        start_byte, sep, last_byte = range_set.partition('-')
```

Comment:
```suggestion
                        start, sep, end = range_set.partition('-')
```

This is the *position* of the first and last byte, not the actual byte